packages/date/src/index.js

```diff
@@ -440,4 +463,49 @@ export function getDate( dateString ) {
 	return momentLib.tz( dateString, WP_ZONE ).toDate();
 }
 
+/**
+ * Creates a moment instance using the given timezone or, if none is provided, using global settings.
+ *
+ * @param {(Date|string|Moment|null)}   dateValue  Date object or string, parsable
+ *                                                 by moment.js.
+ * @param {(string|number|null)}        timezone   Timezone to output result in or
+ *                                                 a UTC offset.
+ *                                                 Defaults to timezone from site.
+ *                                                 See momentjs.com.
+ * @return {Moment}  a moment instance.
+ */
+function buildMoment( dateValue, timezone ) {
+	if ( timezone && ! isUtcOffset( timezone ) ) {
+		return momentLib( dateValue ).tz( timezone );
+	}
+
+	if ( timezone && isUtcOffset( timezone ) ) {
+		return momentLib( dateValue ).utcOffset( timezone );
+	}
+
+	if ( settings.timezone.string ) {
+		return momentLib( dateValue ).tz( settings.timezone.string );
+	}
+
+	return momentLib( dateValue ).utcOffset( settings.timezone.offset );
```

Comment:
Minor: Since every one of these logic branches operates on an instance of moment, it could be pulled out as common:

```suggestion
	const dateMoment = momentLib( dateValue );
	if ( timezone && ! isUtcOffset( timezone ) ) {
		return dateMoment.tz( timezone );
	}

	if ( timezone && isUtcOffset( timezone ) ) {
		return dateMoment.utcOffset( timezone );
	}

	if ( settings.timezone.string ) {
		return dateMoment.tz( settings.timezone.string );
	}

	return dateMoment.utcOffset( settings.timezone.offset );
```

I don't think it makes a big difference, but the motivations could be:

- Makes it clearer that the conditions are each transforming the same common Moment instance in distinct ways.
- It may yield (marginally) smaller code.

Reply:
Fixed in 4ce862d.